[github] clarify the `Describe your test plan` section. (#7958)
DataDog/datadog-agent: .github/PULL_REQUEST_TEMPLATE.md

Anything else we should know when reviewing?

### Describe how to test your change

> Write there any instructions on how you have tested your change

Reviewer: ```suggestion
Write here any instructions on how you have tested your change
```

> Write here any instructions on how you have tested your changes
> and details that should be tested during the QA.

Reviewer 2: Should be the other way around I think:

```suggestion
Write here in detail how you have tested your changes
and instructions on how this should be tested in QA.
```

Author: SGTM.
